x/migration/types/messages_morse_account_state.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ sdk.Msg = &MsgCreateMorseAccountState{}
+
+func NewMsgCreateMorseAccountState(authority string, morseAccountState MorseAccountState) *MsgCreateMorseAccountState {
+	return &MsgCreateMorseAccountState{
+		Authority:         authority,
+		MorseAccountState: morseAccountState,
+	}
+}
+
+func (msg *MsgCreateMorseAccountState) ValidateBasic() error {
```

Comment:
Do we not need to validate morseAccountState?

Reply:
Did you have something specific in mind? I've add a hash comparison in `#ValidateBasic()`, but I'm reluctant to add logic that requires iterating over all accounts. 

We should be able to assume the validity because it's generated from the new migrate sub-command and should be deterministically reproducible **and should have social consensus prior to being created on-chain in Shannon**.

Reply:
@bryanchriswhite, do you mean that the `morseAccountStateHash` is not re-computed onchain?

While not mandatory for the reasons you mentioned, I still think it's a good final check for the msg acceptance and avoidance of any data corruption.